frontend/packages/kubevirt-plugin/src/components/modals/cdrom-vm-modal/cdrom-row.tsx

```diff
@@ -0,0 +1,202 @@
+import * as React from 'react';
+import { MinusCircleIcon } from '@patternfly/react-icons';
+import { getName } from '@console/shared';
+import {
+  FormSelect,
+  FormGroup,
+  FormSelectOption,
+  TextInput,
+  Button,
+  InputGroup,
+} from '@patternfly/react-core';
+import {
+  sourceDict,
+  CD_TYPE_PVC,
+  CD_TYPE_CONTAINER,
+  CD_TYPE_URL,
+  CD_SIZE,
+  CD_STORAGE_CLASS,
+  CD_TYPE_WINTOOLS,
+  WINTOOLS_CONTAINER_NAMES,
+} from './constants';
+
+export const CDRomRow = ({
+  cd,
+  pvcs,
+  storageClasses,
+  index,
+  onChange,
+  onEject,
+  onDelete,
+  isWindows,
+}) => {
+  const {
+    name,
+    type,
+    container,
+    pvc,
+    url,
+    windowsTools,
+    size,
+    storageClass,
+    isURLValid,
+    ejected,
+    isInVM,
+  } = cd;
+  const { upstream } = WINTOOLS_CONTAINER_NAMES;
+
+  if (isInVM && !ejected)
+    return (
+      <div className="kv-cdrom-modal-row">
+        <div className="kv-cdrom-modal-row-name kv-cdrom-modal-padding">{`Drive ${index + 1}: ${
+          sourceDict[type]
+        }`}</div>
+        <InputGroup className="kv-cdrom-modal-padding">
+          <TextInput
+            isDisabled
+            type="text"
+            id="cdrom-value"
```

Comment:
There can be multiple `CDRomRow` components per page. The `id` should be unique.